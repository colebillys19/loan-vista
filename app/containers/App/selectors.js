import { createSelector } from 'reselect';

const selectRouter = (state) => state.router;

const makeSelectPathname = () =>
  createSelector(
    selectRouter,
    (routerState) => routerState.location.pathname,
  );

export { makeSelectPathname };
