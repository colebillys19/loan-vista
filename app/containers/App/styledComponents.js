import styled from 'styled-components';

export const MainWrapper = styled.main`
  min-height: calc(100vh - 14.3rem);
  margin: 0 auto;
  width: 120rem;
`;
