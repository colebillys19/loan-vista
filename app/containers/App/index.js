/**
 * App.js
 * @description ...
 */

import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { push } from 'connected-react-router';

import GlobalStyles from 'globalStyles';

import Calls from 'containers/Calls';
import Documents from 'containers/Documents';
import Main from 'containers/Main';
import Payments from 'containers/Payments';

import DashboardView from 'components/DashboardView';
import AppHeader from 'components/AppHeader';
import MainContainer from 'components/MainContainer';
import MiscView from 'components/MiscView';
import NotFoundView from 'components/NotFoundView';
import SidebarView from 'components/SidebarView';
import TabNav from 'components/TabNav';

import { makeSelectPathname } from './selectors';

const App = ({ dispatchNavigation, pathname }) => (
  <Fragment>
    <AppHeader />
    <TabNav dispatchNavigation={dispatchNavigation} pathname={pathname} />
    <Main
      render={({
        error,
        loading,
        loanNumber,
        sidebarHeaderData,
        sidebarSummariesData,
      }) => (
        <MainContainer>
          <SidebarView
            error={error}
            loading={loading}
            loanNumber={loanNumber}
            pathname={pathname}
            sidebarHeaderData={sidebarHeaderData}
            sidebarSummariesData={sidebarSummariesData}
          />
          <Switch>
            <Route
              exact
              path="/"
              render={() => (
                <DashboardView
                  error={error}
                  loading={loading}
                  pathname={pathname}
                />
              )}
            />
            <Route component={Calls} exact path="/calls" />
            <Route component={Documents} exact path="/documents" />
            <Route component={Payments} exact path="/payments" />
            <Route
              exact
              path="/misc"
              render={() => (
                <MiscView error={error} loading={loading} pathname={pathname} />
              )}
            />
            <Route component={NotFoundView} />
          </Switch>
        </MainContainer>
      )}
    />
    <GlobalStyles />
  </Fragment>
);

App.propTypes = {
  dispatchNavigation: T.func.isRequired,
  pathname: T.string.isRequired,
};

const mapStateToProps = createStructuredSelector({
  pathname: makeSelectPathname(),
});

const mapDispatchToProps = (dispatch) => ({
  dispatchNavigation: (pathname) => dispatch(push(pathname)),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(App);
