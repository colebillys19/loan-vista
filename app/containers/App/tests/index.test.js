import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';

import configureStore from 'configureStore';
import history from 'utils/history';

import App from '../index';

// Create redux store with history
const initialState = {};
const store = configureStore(initialState, history);

const WrappedApp = (props) => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <App {...props} />
    </ConnectedRouter>
  </Provider>
);

const renderer = new ShallowRenderer();

describe('<App />', () => {
  it.skip('Should render and match the snapshot', () => {
    renderer.render(<WrappedApp />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
