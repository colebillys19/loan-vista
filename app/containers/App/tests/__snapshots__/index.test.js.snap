// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Memo(ConnectFunction)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Memo(ConnectFunction) />
  </Memo(ConnectFunction)>
</Provider>
`;
